src/common/lpuart.rs

```diff
@@ -930,6 +930,88 @@ impl<P, const N: u8> eh02::blocking::serial::Write<u8> for Lpuart<P, N> {
     }
 }
 
+impl eio06::Error for ReadFlags {
+    fn kind(&self) -> eio06::ErrorKind {
+        eio06::ErrorKind::Other
+    }
+}
+
+impl<P, const N: u8> eio06::ErrorType for Lpuart<P, N> {
+    type Error = ReadFlags;
+}
+
+impl<P, const N: u8> eio06::WriteReady for Lpuart<P, N> {
+    fn write_ready(&mut self) -> Result<bool, Self::Error> {
+        Ok(self.status().contains(Status::TRANSMIT_EMPTY))
+    }
+}
+
+impl<P, const N: u8> eio06::Write for Lpuart<P, N> {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, Self::Error> {
+        use eio06::WriteReady;
+
+        // Block until we can write.
+        // In combination with `WriteReady` this call can be non-blocking.
+        while !self.write_ready()? {}
+
+        let mut num_written = 0;
+        for word in buf {
+            self.write_byte(*word);
+            num_written += 1;
+
+            if !self.write_ready()? {
```

Comment:
I take it there's no contractual requirement to write all the bytes for this call?

Reply:
No, not at all. The contract says "blocking until at least one is written"; same for 'read'.

Reply:
@teburd The trait provides a [`write_all`](https://docs.rs/embedded-io/latest/src/embedded_io/lib.rs.html#394-403) function that wraps around it if you want the contractual guarantee.